src/java/org/apache/cassandra/cql3/QueryProcessor.java

```diff
@@ -674,14 +678,14 @@ public static ResultMessage.Prepared storePreparedStatement(String queryString,
     {
         // Concatenate the current keyspace so we don't mix prepared statements between keyspace (#5352).
         // (if the keyspace is null, queryString has to have a fully-qualified keyspace so it's fine.
-        long statementSize = ObjectSizes.measureDeep(prepared.statement);
+        MD5Digest statementId = computeId(queryString, keyspace);
         // don't execute the statement if it's bigger than the allowed threshold
-        if (statementSize > capacityToBytes(DatabaseDescriptor.getPreparedStatementsCacheSizeMB()))
+        if ((prepared.pstmntSize + statementId.size()) > capacityToBytes(DatabaseDescriptor.getPreparedStatementsCacheSizeMB()))
```

Comment:
we might use  `getSizeForCache` for `(prepared.pstmntSize + statementId.size()`

Reply:
+1